docs/01_cva6_user/Traps_Interrupts_Exceptions.rst

```diff
@@ -1,25 +1,126 @@
-
-   SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
-
-.. Level 1
-   =======
-
-   Level 2
-   -------
-
-   Level 3
-   ~~~~~~~
-
-   Level 4
-   ^^^^^^^
-
-.. _cva6_traps_interrupts_exceptions:
-
-Traps, Interrupts, Exceptions
-=============================
-Gap for step1 verification. Reuse DVplan from E4? Jean-Roch will check with André. Mike will reach out to some people who could help.
-We expect this section to be 1 page.
-
+﻿..
+   Copyright (c) 2023 OpenHW Group
+   Copyright (c) 2023 Thales DIS SAS
+
+   SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
+
+.. Level 1
+   =======
+
+   Level 2
+   -------
+
+   Level 3
+   ~~~~~~~
+
+   Level 4
+   ^^^^^^^
+
+.. _cva6_traps_interrupts_exceptions:
+
+Traps, Interrupts, Exceptions
+=============================
+Traps are composed of interrupts and exceptions.
+Interrupts are asynchronous events whereas exceptions are synchronous ones.
+On one hand, interrupts are occuring independently of the instructions
+(mainly raised by peripherals or debug module).
+On the other hand, an instruction may raise exceptions synchronously.
+
+Raising Traps
+=============
+When a trap is raised, the behaviour of the CVA6 core depends on
+several CSRs and some CSRs are modified. The CSR description is available
+in :doc:`CV32A6_Control_Status_Registers.rst`.
+
+Configuration CSRs
+------------------
+CSRs having an effect on the core behaviour when a trap occurs are:
+
+* ``mstatus`` and ``sstatus``: several fields are read like interrupt enable (MIE, SIE)
+* ``mtvec`` and ``stvec``: specifies the address of trap handler.
+* ``medeleg``: specifies which exceptions can be handled by a lower privileged mode
+* ``mideleg``: specifies which interrupts can be handled by a lower privileged mode
+
+Modified CSRs
+-------------
+* ``mstatus`` or ``sstatus``: several fields are updated like previous privilege mode (MPP, SPP), previous interrupt enabled (MPIE, SPIE)
+* ``mepc`` or ``sepc``: updated with the virtual address of the interrupted instruction or the instruction raising the exception.
+* ``mcause`` or ``scause``: updated with a code indicating the event causing the trap.
+* ``mtval`` or ``stval``: updated with exception specific information like the faulting virtual address
+
+Supported exceptions
+--------------------
+The following exceptions are supported by the CVA6:
+
+* instruction address misaligned
+
+  * control flow instruction with misaligned target
+
+* instruction access fault
+
+  * physical address fault
+
+* illegal instruction:
+
+  * unimplemented CSRs
+  * unsupported extensions
+
+* breakpoint
+* load address misaligned:
+
+  * ``LH`` at 2n+1 address
+  * ``LW`` at 4n+1, 4n+2, 4n+3 address
+
+* load access fault
+
+  * access to PMP region without read permissions
+
+* store/AMO address misaligned
+
+  * ``SH`` at 2n+1 address
+  * ``SW`` at 4n+1, 4n+2, 4n+3 address
+
+* store/AMO access fault
+
+  * access to PMP region without write permissions
```

Comment:
What about write and execute permissions?

Reply:
For store, only write permissions